### ISSUE TITLE
`decompose_toffoli=2` throws QiskitError

### ISSUE THREAD

**Issue by teaguetomesh** [Quantum-Software-Tools/quantum-constrained-optimization#2]
When attempting to generate QAO-Ansatz or DQVA circuits with `decompose_toffoli=2` (should decompose the multi-controlled Rx gates to IBM's basis set {u1, u2, u3, CX}), we get the error 

``` python
QiskitError: "Error decomposing node of instruction 'p': 'NoneType' object has no attribute 'global_phase'. Unable to define instruction 'u' in the given basis."
```

**Comment by nicholas-d-allen** [comment]
Just attempted to solve the problem using the `BasisTranslator` class, specifically by using Qiskit's built-in `StandardEquivalenceLibrary` by changing the `Unroller` pass with the following:
```python
bt_pass = BasisTranslator(StandardEquivalenceLibrary, basis_gates)
pm = PassManager(bt_pass)
qaoa_circ = pm.run(qaoa_circ)
```
However, this did not solve the problem with decompiling the QAO-Ansatz circuits, and instead throws this error:
```python
TranspilerError: "Unable to map source basis {('mcx', 4), ('mcx', 5), ('barrier', 8), ('x', 1), ('ry', 1), ('rz', 1), ('ccx', 3)} to target basis {'reset', 'snapshot', 'u3', 'u1', 'cx', 'u2', 'measure', 'barrier', 'delay'} over library <qiskit.circuit.equivalence.EquivalenceLibrary object at 0x7f33ab921b70>."
```
Normally, this error occurs when [the Transpiler is unable to reach a certain basis](https://github.com/Qiskit/qiskit-terra/issues/5539#issue-769353876), however we know from the "adding `'u'` solution" that this basis should be reachable. 

The underlying issue with using the built in `StandardEquivalenceLibrary` with the `BasisTranslator` is that it is unable to decompose large multi-controlled NOT gates beyond just CCNOTs. As a MWE, consider the following code which decomposes a CCNOT gate (adapted from [this SE post](https://stackoverflow.com/questions/66881062/how-to-get-a-default-equivalencelibrary)):
```python
basis_gates = ['u1', 'u2', 'u3', 'cx']
bt_pass = BasisTranslator(StandardEquivalenceLibrary, basis_gates)

circuit = QuantumCircuit(3)
circuit.mcx([0,1],2)

dag_out = bt_pass.run(circuit_to_dag(circuit))
circuit_out = dag_to_circuit(dag_out)
circuit_out.draw(fold=-1)
```
Which outputs
```
                                                                                  ┌─────────┐      
q_0: ──────────────────────────────■────────────────────────────────■───────■─────┤ U1(π/4) ├───■──
                                   │                  ┌─────────┐   │     ┌─┴─┐   ├─────────┴┐┌─┴─┐
q_1: ─────────────■────────────────┼───────────────■──┤ U1(π/4) ├───┼─────┤ X ├───┤ U1(-π/4) ├┤ X ├
     ┌─────────┐┌─┴─┐┌──────────┐┌─┴─┐┌─────────┐┌─┴─┐├─────────┴┐┌─┴─┐┌──┴───┴──┐├─────────┬┘└───┘
q_2: ┤ U2(0,π) ├┤ X ├┤ U1(-π/4) ├┤ X ├┤ U1(π/4) ├┤ X ├┤ U1(-π/4) ├┤ X ├┤ U1(π/4) ├┤ U2(0,π) ├──────
     └─────────┘└───┘└──────────┘└───┘└─────────┘└───┘└──────────┘└───┘└─────────┘└─────────┘      
```
However, replacing the CCNOT with a C(3)NOT:
```python
...
circuit = QuantumCircuit(4)
circuit.mcx([0,1,2],3)
...
```
reproduces the same error:
```python
TranspilerError: "Unable to map source basis {('mcx', 4)} to target basis {'reset', 'snapshot', 'u3', 'cx', 'u1', 'u2', 'measure', 'barrier', 'delay'} over library <qiskit.circuit.equivalence.EquivalenceLibrary object at 0x7f33ab921b70>."
```
By visiting [the source code for the `StandardEquivalenceLibrary`,](https://github.com/Qiskit/qiskit-terra/blob/0.16.4/qiskit/circuit/library/standard_gates/equivalence_library.py) it becomes obvious that MCX gate decomposition is indeed not supported. 

As such, for this specific issue, I would suggest the short term solution of sticking with the `Unroller` and just replacing the gate set to include `'u'`. A more permanent, long-term solution would be to implement MCX gate support for the `StandardEquivalenceLibrary`, but that's a separate project on its own.

**Comment by teaguetomesh** [comment]
Closed by #3
